src/coreComponents/dataRepository/xmlWrapper.cpp

```diff
@@ -31,13 +31,98 @@ using namespace dataRepository;
 namespace xmlWrapper
 {
 
+void validateString( string const & value, Regex const & regex )
+{
+  std::smatch m;
+  bool inputValidated = std::regex_search( value, m, std::regex( regex.m_regexStr ) );
+  if( !inputValidated || m.length() != ptrdiff_t( value.length() ) )
+  {
+    ptrdiff_t errorId = ( m.size()>0 && m.position( 0 )==0 ) ? m.length() : 0;
+    GEOS_THROW( "Input string validation failed at:\n" <<
+                "  \"" << value << "\"\n   " << string( errorId, ' ' ) << "^\n" <<
+                "  Expected format: " << regex.m_formatDescription,
```

Comment:
Suggestion: good use case for `GEOS_FMT` if you can figure out how (you can use a runtime argument to specify padding width, we do it somewhere in GEOS, probably in VTK output)

Reply:
After searching what is possible with both `fmt::format` and `std::format`, I end up with this:
```c++
    GEOS_THROW( GEOS_FMT( "Input string validation failed at:\n"
                          "  \"{}\"\n"
                          "   {:>{}}\n"
                          "  Expected format: {}",
                          value, "^", errorId+1, myRegex.m_formatDescription ) );
```

Reply:
Yes, exactly what I had in mind!